contrib/emicklei/go-restful/restful_test.go

```diff
@@ -0,0 +1,154 @@
+package restful
+
+import (
+	"errors"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/emicklei/go-restful"
+	"github.com/stretchr/testify/assert"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/mocktracer"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+)
+
+func TestChildSpan(t *testing.T) {
+	assert := assert.New(t)
+	mt := mocktracer.Start()
+	defer mt.Stop()
+
+	ws := new(restful.WebService)
+	ws.Filter(Trace)
+	ws.Route(ws.GET("/user/{id}").To(func(request *restful.Request, response *restful.Response) {
+		_, ok := tracer.SpanFromContext(request.Request.Context())
+		assert.True(ok)
+	}))
+
+	container := restful.NewContainer()
+	container.Add(ws)
+
+	r := httptest.NewRequest("GET", "/user/123", nil)
+	w := httptest.NewRecorder()
+
+	container.ServeHTTP(w, r)
+}
+
+func TestTrace200(t *testing.T) {
```

Comment:
I'm not sure if this is possible (in a nice way), but do you think we could look into potentially having less tests and more sub-tests?

Reply:
My understanding of the purpose of subtests is to combine variations of input to a function being tested.
Are you thinking something like
```go
// setup tracer & webservice
...
// define multiple routes here
testCases := []struct{ ... }
// register multiple routes here
for ( ... ) {
  // endpoint handler with assertions
   ws.route( ... )
} 

// run each route
for ( ... ) {
  ...
   container.ServeHTTP( ... )
  ...
} 
```

This probably will result in single long test function.

Reply:
Not quite, I think it can be a lot simpler than your example. Maybe you can check the other integrations to get an idea. The result should be simpler than what we have now. If it isn't, then let's leave it in the current state.